src/src/tx_main.cpp

```diff
@@ -389,6 +390,47 @@ void ICACHE_RAM_ATTR HandlePrepareForTLM()
   }
 }
 
+int8_t ICACHE_RAM_ATTR PowerEnumToLBTLimit(PowerLevels_e txPower)
+{
+    // Calculated from EN 300 328, assuming 800kHz BW for sx1280
+    // TL = -70 dBm/MHz + 10 × log10 (100 mW / Pout) (Pout in mW e.i.r.p.)
+    // Values above 100mW are not relevant, default to 100mW threshold
+    // TODO: This threshold should be modified with a config for antenna gain
+    switch(txPower)
+    {
+    case PWR_10mW: return -78;
+    case PWR_25mW: return -82;
+    case PWR_50mW: return -84;
+    case PWR_100mW: return -88;
+    default: return -88;
+    }
+}
+
+volatile uint32_t RxStartTimeCCA = 0;
+
+void ICACHE_RAM_ATTR BeginClearChannelAssessment()
+{
+  // Listen Before Talk (LBT) aka clear channel assessment (CCA)
+  // Start RX early because it takes about 90us from RX enable to 
+  // valid instant RSSI values are returned.
+  // Not interested in packets or interrupts while measuring RF energy on channel.
+
+  Radio.SetDioIrqParams(SX1280_IRQ_RADIO_NONE, SX1280_IRQ_RADIO_NONE, SX1280_IRQ_RADIO_NONE, SX1280_IRQ_RADIO_NONE);
+  Radio.RXnb();
+  RxStartTimeCCA = micros();
+}
+
+bool ICACHE_RAM_ATTR ChannelIsClear()
+{
+  while(micros() - RxStartTimeCCA < ExpressLRS_currAirRate_RFperfParams->InstantRssiValidDelayUs);
+  int8_t rssiResult = Radio.GetRssiInst();
```

Comment:
Should this be called in a while loop for 18us?

Reply:
Radio.GetRssiInst() gives the RSSI averaged over last symbol-time and takes about minimum 100us to give valid value from RX is enabled. So as long as minimum LBT scan time is less than symbol time, which it is for all modes I think, it should be enough to call it once as soon as the result is ready?

Reply:
Now the latest commit enables rx much earlier, in tx interrupt. Calling getRssiInst() once should be OK, since it will give average rssi over last symbol-time, which is always longer than the required listening window.